portfolio/src/main/webapp/script.js

```diff
@@ -86,40 +86,55 @@ function swapMotorcyclePicture() {
 }
 
 // Delete Messages
-function deleteMessages() {
-  const request = new Request('/delete-data', {method: 'POST'});
-  fetch(request).then(() => getMessages());
+function deleteMessage(id) {
+  const params = new URLSearchParams();
+  params.append('id', id);
+  fetch('/delete-data', {method: 'POST', body: params}).then(() => getMessages());
 }
 
 // Upload message to website
 function postMessage() {
+  const textInput = document.getElementById('text-input');
   const name = document.getElementById('name-input').value;
-  const text = document.getElementById('text-input').value;
-  const request = new Request('/data', {method: 'POST', body: name + '*' + text});
-  fetch(request).then(() => getMessages());
+  const text = textInput.value;
+  if (text != "" & name != ""){
+    const params = new URLSearchParams();
+    params.append('name-input', name);
+    params.append('text-input', text);
+    const request = new Request('/data', {method: 'POST', body: params});
+    fetch(request).then(() => getMessages());
+  }
+  textInput.value = "";
 }
 
 // send GET request to server let
 function getMessages() {
   fetch('/data').then(response => response.json()).then( messages => {
+    const height = Math.min(messages.length, 5) * 100
     const displayMessages = document.getElementById('messages-container');
     displayMessages.innerHTML = "";
     messages.forEach(message => displayMessages.appendChild(
-      createMessage(message.val0, message.val1)));
```

Comment:
correction to my above discussion of whether to put this in a class or not - definitely put message in a class. that will allow the JSON serialized format to include type names, so you will be able to change this to be message.name, message.id, etc.

Reply:
okay